src/ser/mod.rs

```diff
@@ -96,20 +95,22 @@ impl<'a> Serializer<'a> {
 // which take 200+ bytes of ROM / Flash
 macro_rules! serialize_unsigned {
     ($self:ident, $N:expr, $v:expr) => {{
-        let mut buf: [u8; $N] = unsafe { super::uninitialized() };
```

Comment:
Can we just make this way simpler by using heapless instead? Would avoid the whole MaybeUninit shenanigans.

Reply:
happy to look into this, any pointers?

Reply:
Here a pointer to the [`heapless`](https://crates.io/crates/heapless) crate.